azerothcore/azerothcore-wotlk
# Fixed hunter pets not being revived inside battlegrounds.

Fixed hunter pets not having max health when player is resurrected in battlegrounds.

Signed-off-by: ChrisCGalbraith <chriscgalbraith@gmai.com>



## Changes Proposed:
-  New summon of the players pet occurs when revived by the battleground spirit healer

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/1193

## SOURCE:


## Tests Performed:

- Performed tests on hunters and warlocks with pets that are dead at the time of resurrection and pets that have a portion of their health missing.
- Built on Windows 10, no errors.


## How to Test the Changes:


1. .debug bg on
2. Enter a battleground and wait for it to begin
3. .die your pet and then yourself
4.  Observe that pet is revived at full health when the player is.
5.  Repeat the above steps but instead of killing your pet, .damage him

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
